pontoon/sync/core/entities.py

```diff
@@ -128,32 +128,33 @@ def update_resources(
         f"[{project.slug}] Changed source files: {', '.join(res.path for res in changed_resources)}"
     )
 
-    prev_entities = {
+    prev_entities: dict[tuple[str, str], Entity] = {
         (e.resource.path, e.key or e.string): e
         for e in Entity.objects.filter(resource__in=changed_resources, obsolete=False)
         .select_related("resource")
         .iterator()
     }
-    next_entities = {
+    next_entities: dict[tuple[str, str], Entity] = {
         (path, entity.key or entity.string): entity
         for path, entity in (
             (cr.path, entity_from_source(cr, now, 0, tx))
             for cr in changed_resources
             for tx in updates[cr.path]
         )
     }
-
-    obsolete_entities = [
-        ent
-        for key, ent in prev_entities.items()
-        if key in prev_entities.keys() - next_entities.keys()
-    ]
-    for ent in obsolete_entities:
-        ent.obsolete = True
-        ent.date_obsoleted = now
-    obs_count = Entity.objects.bulk_update(
-        obsolete_entities, ["obsolete", "date_obsoleted"]
-    )
+    prev_not_next = prev_entities.keys() - next_entities.keys()
+    prev_and_next = prev_entities.keys() & next_entities.keys()
+    next_not_prev = next_entities.keys() - prev_entities.keys()
+
+    obsolete_entities: list[Entity] = []
+    log_rm: dict[str, list[str]] = defaultdict(list)
+    for key, prev_ent in prev_entities.items():
+        if key in prev_not_next:
+            prev_ent.obsolete = True
+            prev_ent.date_obsoleted = now
+            obsolete_entities.append(prev_ent)
+            log_rm[key[0]].append(key[1].replace("\n", "\\n"))
```

Comment:
Something like this would be more readbale:

```suggestion
            log_rm[path].append(entity.replace("\n", "\\n"))
```